config/config.go

```diff
@@ -1157,6 +1122,91 @@ func migrateConfigTCF2PurposeEnabledFlags(v *viper.Viper) {
 	}
 }
 
+func migrateConfigDatabaseConnection(v *viper.Viper) {
+
+	migrations := []struct {
+		old    string
+		new    string
+		fields []string
+	}{
+		{
+			old: "stored_requests.postgres",
+			new: "stored_requests.database",
+			fields: []string{
+				"connection.dbname",
+				"connection.host",
+				"connection.port",
+				"connection.user",
+				"connection.password",
+				"fetcher.query",
+				"fetcher.amp_query",
+				"initialize_caches.timeout_ms",
+				"initialize_caches.query",
+				"initialize_caches.amp_query",
+				"poll_for_updates.refresh_rate_seconds",
+				"poll_for_updates.timeout_ms",
+				"poll_for_updates.query",
+				"poll_for_updates.amp_query",
+			},
+		},
+		{
+			old: "stored_video_req.postgres",
+			new: "stored_video_req.database",
+			fields: []string{
+				"connection.dbname",
+				"connection.host",
+				"connection.port",
+				"connection.user",
+				"connection.password",
+				"fetcher.query",
+				"initialize_caches.timeout_ms",
+				"initialize_caches.query",
+				"poll_for_updates.refresh_rate_seconds",
+				"poll_for_updates.timeout_ms",
+				"poll_for_updates.query",
+			},
+		},
+		{
+			old: "stored_responses.postgres",
+			new: "stored_responses.database",
+			fields: []string{
+				"connection.dbname",
+				"connection.host",
+				"connection.port",
+				"connection.user",
+				"connection.password",
+				"fetcher.query",
+				"initialize_caches.timeout_ms",
+				"initialize_caches.query",
+				"poll_for_updates.refresh_rate_seconds",
+				"poll_for_updates.timeout_ms",
+				"poll_for_updates.query",
+			},
+		},
+	}
+
+	for _, migration := range migrations {
+		driverField := migration.new + ".connection.driver"
+		if !v.IsSet(driverField) && v.IsSet("postgres.connection.dbname") {
```

Comment:
I tested it locally with current config and found this statement `v.IsSet("postgres.connection.dbname")` returns false, because this is not a full config name. Should be `migration.old + ".connection.dbname"`. 
Then it start to work correctly. I would suggest to modify this `if` condition to this:
```
if !v.IsSet(driverField) && strings.Contains(migration.old, "postgres") {
```

Reply:
Yes, right, this works too!

Reply:
Updated as proposed in the comment below.